openedx/core/djangoapps/course_groups/views.py

```diff
@@ -379,3 +390,356 @@ def debug_cohort_mgmt(request, course_key_string):
+
+        * cohorts: List of cohorts.
+        * cohort: A cohort representation:
+            * name: The string identifier for a cohort.
+            * id: The integer identifier for a cohort.
+            * user_count: The number of students in the cohort.
+            * assignment_type: The string representing the assignment type.
+            * user_partition_id: The integer identified of the UserPartition.
+            * group_id: The integer identified of the specific group in the partition.
+    """
+
+    def get(self, request, course_key_string, cohort_id=None):
+        """
+        Endpoint to get either one or all cohorts.
+        """
+        course_key, course = _get_course_with_access(request, course_key_string)
+        if not cohort_id:
+            all_cohorts = cohorts.get_course_cohorts(course)
+            paginator = NamespacedPageNumberPagination()
+            paginator.max_page_size = MAX_PAGE_SIZE
+            page = paginator.paginate_queryset(all_cohorts, request)
+            response = [_get_cohort_representation(c, course) for c in page]
+            return Response(response, status=status.HTTP_200_OK)
+        else:
+            cohort = cohorts.get_cohort_by_id(course_key, cohort_id)
+            return _get_cohort_response(cohort, course)
+
+    def post(self, request, course_key_string, cohort_id=None):
+        """
+        Endpoint to create a new cohort, must not include cohort_id.
+        """
+        if cohort_id is not None:
+            raise self.api_error(status.HTTP_405_METHOD_NOT_ALLOWED,
+                                 'Please use the parent endpoint.',
+                                 'wrong-endpoint')
+        course_key, course = _get_course_with_access(request, course_key_string)
+        name = request.data.get('name')
+        if not name:
+            raise self.api_error(status.HTTP_400_BAD_REQUEST,
+                                 '"name" must be specified to create cohort.',
+                                 'missing-cohort-name')
+        assignment_type = request.data.get('assignment_type')
+        if not assignment_type:
+            raise self.api_error(status.HTTP_400_BAD_REQUEST,
+                                 '"assignment_type" must be specified to create cohort.',
+                                 'missing-assignment-type')
+        return _get_cohort_response(
+            cohorts.add_cohort(course_key, name, assignment_type), course)
+
+    def patch(self, request, course_key_string, cohort_id=None):
+        """
+        Endpoint to update a cohort information, including:
+        """
+        if cohort_id is None:
+            raise self.api_error(status.HTTP_405_METHOD_NOT_ALLOWED,
+                                 'Request method requires cohort_id in path',
+                                 'missing-cohort-id')
+        name = request.data.get('name')
+        assignment_type = request.data.get('assignment_type')
+        if {None} == {name, assignment_type}:
```

Comment:
This looks like a strange idiom. You could also say ``if not any((name, assignment_type))``

Reply:
The PR has been updated to use the suggested condition.